native/src/main/java/io/ballerina/stdlib/rabbitmq/util/ListenerUtils.java

```diff
@@ -167,16 +169,38 @@ public static Object detach(Environment environment, BObject listenerBObject, BO
     private static void declareQueueIfNotExists(BObject service, Channel channel) throws IOException {
         BMap serviceConfig = (BMap) ((AnnotatableType) TypeUtils.getType(service))
                 .getAnnotation(StringUtils.fromString(ModuleUtils.getModule().getOrg() + ORG_NAME_SEPARATOR
-                                                              + ModuleUtils.getModule().getName() + VERSION_SEPARATOR
-                                                              + ModuleUtils.getModule().getVersion() + ":"
-                                                              + RabbitMQConstants.SERVICE_CONFIG));
+                        + ModuleUtils.getModule().getName() + VERSION_SEPARATOR
+                        + ModuleUtils.getModule().getVersion() + ":"
+                        + RabbitMQConstants.SERVICE_CONFIG));
         String queueName;
         if (service.getNativeData(RabbitMQConstants.QUEUE_NAME.getValue()) != null) {
+            // Queue name is given as the service name
             queueName = (String) service.getNativeData(RabbitMQConstants.QUEUE_NAME.getValue());
         } else {
+            // Queue name and config given in the service config
             queueName = serviceConfig.getStringValue(RabbitMQConstants.QUEUE_NAME).getValue();
+            Map<String, Object> argumentsMap = new HashMap<>();
+            boolean durable = false;
+            boolean exclusive = false;
+            boolean autoDelete = true;
+
+            if ((BMap<BString, Object>) serviceConfig.getMapValue(RabbitMQConstants.QUEUE_CONFIG) != null) {
+
+                @SuppressWarnings(RabbitMQConstants.UNCHECKED)
+                BMap<BString, Object> queueConfig =
+                        (BMap<BString, Object>) serviceConfig.getMapValue(RabbitMQConstants.QUEUE_CONFIG);
+                durable = queueConfig.getBooleanValue(RabbitMQConstants.QUEUE_DURABLE);
+                exclusive = queueConfig.getBooleanValue(RabbitMQConstants.QUEUE_EXCLUSIVE);
+                autoDelete = queueConfig.getBooleanValue(RabbitMQConstants.QUEUE_AUTO_DELETE);
+                if (queueConfig.getMapValue(RabbitMQConstants.QUEUE_ARGUMENTS) != null) {
```

Comment:
```suggestion
                if (queueConfig.containsKey(RabbitMQConstants.QUEUE_ARGUMENTS)) {
```

Would this work here ?

Reply:
i think the default value is null so the key will be there with null as value.